common/include/pcl/common/impl/centroid.hpp

```diff
@@ -269,7 +279,7 @@ computeCovarianceMatrix (const pcl::PointCloud<PointT> &cloud,
     return (0);
 
   // Initialize to 0
-  covariance_matrix.setZero ();
+  Eigen::Matrix<Scalar, 3, 3> accu = Eigen::Matrix<Scalar, 3, 3>::Zero ();
```

Comment:
I wonder if using `selfAdjointView` would make sense here

Reply:
do you mean something like `accu = accu.selfadjointView<Eigen::Upper>();` after filling in the upper triangle? (i.e. instead of the three lines which copy to the lower triangle)
Since it's just a 3x3 matrix, I'm not sure whether we gain much, but I can add that change if you think it makes sense.

Reply:
It was more like: is the added complexity due to selfAdjointView a good trade-off for making it explicit? Performance wise, both will be the same

Reply:
on first glance, I think the explicit version is easier to understand